Tests/CacheAdvanceTests/CacheAdvanceTests.swift

```diff
@@ -0,0 +1,391 @@
+        "Phasellus convallis velit quis viverra pellentesque.",
+        "Duis sit amet laoreet nunc.",
+        "Vestibulum magna odio,",
+        "aliquam feugiat urna quis,",
+        "interdum condimentum sapien.",
+        "Donec varius ipsum non mattis hendrerit.",
+        "Fusce a laoreet ligula.",
+        "Cras efficitur posuere ante quis ullamcorper.",
+        "Donec ut varius quam,",
+        "sit amet bibendum ipsum.",
+        "Proin molestie,",
+        "nulla blandit hendrerit laoreet,",
+        "erat sapien mattis odio,",
+        "eu egestas erat est id nulla.",
+        "Integer pulvinar feugiat justo a mollis.",
+        "Maecenas nisi nisl,",
+        "lacinia eget convallis eu,",
+        "hendrerit sit amet quam.",
+        "Vestibulum mattis velit eu sapien maximus pellentesque.",
+        "Vivamus venenatis,",
+        "ex at condimentum mollis,",
+        "odio turpis elementum dui,",
+        "sed accumsan odio sem a nibh.",
+        "Suspendisse sed tincidunt urna,",
+        "quis aliquam risus.",
+        "Maecenas vitae lacinia ante.",
+        "Nulla quis est mi.",
+        "Nunc non maximus nulla.",
+        "Phasellus placerat elit ac pretium pharetra.",
+        "Nunc nibh dolor,",
+        "convallis non ultrices in,",
+        "pharetra a massa.",
+        "In hac habitasse platea dictumst.",
+        "Integer mattis luctus metus,",
+        "eget pretium elit semper a.",
+        "In interdum congue nibh vel porttitor.",
+        "Phasellus eu viverra turpis,",
+        "ut molestie metus.",
+        "Suspendisse quis eros mollis,",
+        "cursus enim in,",
+        "malesuada diam.",
+        "Nullam in metus vulputate,",
+        "finibus nisi ut,",
+        "pellentesque tortor.",
+        "Mauris rutrum,",
+        "lectus ullamcorper elementum dignissim,",
+        "orci neque condimentum dolor,",
+        "quis tempus ante urna ac dui.",
+        "Vestibulum dui elit,",
+        "pulvinar at velit non,",
+        "maximus semper tortor.",
+        "Ut eu neque sit amet nulla aliquet commodo nec fermentum purus.",
+        "Mauris ut urna a est sollicitudin condimentum id in enim.",
+        "Aliquam porttitor libero id laoreet placerat.",
+        "Etiam euismod libero eget risus placerat,",
+        "quis egestas sapien lacinia.",
+        "Donec eget augue dignissim,",
+        "ultrices elit eget,",
+        "dictum nibh.",
+        "In ultricies risus vel nisi convallis fermentum.",
+        "Etiam tempor nisi nulla,",
+        "eu pulvinar nisl pretium ut.",
+        "Cras ullamcorper enim nisl,",
+        "at tempus arcu sagittis quis.",
+    ]
+
+    private func requiredByteCount<T: Codable>(for messages: [T], cacheWillRoll: Bool) throws -> UInt64 {
+        let encoder = JSONEncoder()
+        let messageSpanSuffixLength = cacheWillRoll ? Bytes(Data.messageSpanStorageLength + Data.bytesStorageLength) : Bytes(Data.messageSpanStorageLength)
```

Comment:
It could be cool if there was one or more `internal` methods we could call on the production code to compute this stuff, rather than doing it ourself here. But not a big deal. Just a thought.

Reply:
Would certainly allow for faster iteration on new file formats

I’ll likely improve this the moment I want to change the file format 😂

In the meantime I’ll save this for a later PR